pywb/warcserver/index/indexsource.py

```diff
@@ -47,24 +56,30 @@ def _init_sesh(self, adapter=None):
 class FileIndexSource(BaseIndexSource):
     CDX_EXT = ('.cdx', '.cdxj')
 
-    def __init__(self, filename):
+    def __init__(self, filename, config=None):
         self.filename_template = filename
 
-    def load_index(self, params):
-        filename = res_template(self.filename_template, params)
-
+    def _do_open(self, filename):
         try:
-            fh = open(filename, 'rb')
+            return open(filename, 'rb')
         except IOError:
             raise NotFoundException(filename)
 
-        def do_load(fh):
+    def load_index(self, params):
+        filename = res_template(self.filename_template, params)
+
+        fh = self._do_open(filename)
+
+        def do_iter():
```

Comment:
Hm, well the goal was to not require the derived `_do_iter` to be concerned with closing `fh`, to ensure that its closed no matter what..  But maybe cleaner to move the with fh into the derived function?

Reply:
I believe it would be cleaner to move the `with fh` to the `_do_iter` method. 

We could also do some refactoring of these classes such that each one has their own version of `_do_iter` but as `__do_iter`.
The double underscore version would be transformed to `_classname__methodName` [docs](https://docs.python.org/3/tutorial/classes.html#private-variables).
This would ensure that we do not have any name clashes in the inheritance chain.

Reply:
Hm, i think as is now is the safest, as don't have to worry about adding a `with fh` into every derived implementation.. Making private doesn't seem to be necessary..